### ISSUE TITLE
Error: "You can not change the userId property"

### ISSUE THREAD

**Issue by jeffijoe** [layerhq/layer-websdk#156]
![image](https://user-images.githubusercontent.com/2505474/26949582-cb0a023a-4c9a-11e7-9cbc-4647d069fd59.png)

I'm trying to implement presence support, and the issue happens due to this:

```
client.getIdentity(`layer:///identities/${encodeURIComponent(user.id)}`, true)
```

If I change it to

```
client.getIdentity(user.id, true)
```

I get the same error.

The error seems to occur when the identity is being populated with values from the server.

**Comment by michaelkantor** [comment]
What happens if you use:
```
client.on('ready', function() {
    client.getIdentity(...);
});
```

**Comment by jeffijoe** [comment]
I already do this, omitted for brevity.

**Comment by michaelkantor** [comment]
What is the user_id you are getting? 
What is the user id returned by `client.user.id`?  
What version of the websdk are you on?

**Comment by jeffijoe** [comment]
* It's something along the lines of `auth0|123`
* I'll get this to you in the morning.
* Latest Web SDK on npm

**Comment by jeffijoe** [comment]
`client.user.id` is `layer:///identities/auth0%7C587908a931c1fe22492b7d02`.

**Comment by michaelkantor** [comment]
The WebSDK is behaving as if
```
layer:///identities/${encodeURIComponent(user.id)} === client.user.id
```
These two being the same doesn't seem to match your understanding of whats going on.

I'm trying to understand this mismatch.  

1. Can you sanity check that the `user.id` your using doesn't match `client.user.userId`?  
2. If that doesn't lead to anything, is there a url/login you can send me to enable me to review this bug in action?

**Comment by jeffijoe** [comment]
So you're saying I need to check `if (user.id !== client.user.userId)` before I try to get the identity? Else I'm not sure I understand.

**Comment by michaelkantor** [comment]
Neither do I.  If you'd like to skip to step 2, go ahead.  If thats not an option, we'll try and break this apart.  First step in breaking this apart: Run with break-on-error enabled, and look at the stack trace when the error is thrown, and see WHAT action is trying to set the user thus causing the error to throw.

**Comment by jeffijoe** [comment]
Either way, here's a screenshot showing all the values:

![image](https://user-images.githubusercontent.com/2505474/27046987-c17204b6-4fa6-11e7-8615-d9a828e24de5.png)

**Comment by jeffijoe** [comment]
You can access our app in staging: https://app.t4xstokers..com

Username: jobsapi@test.com
Password: 123456

The error occurs right after logging in.

Some context: Whenever I get another user into my memory store, I set up some handlers that will sync the identity status onto my own models.

**Comment by michaelkantor** [comment]
https://github.com/layerhq/layer-websdk/blob/master/src/models/identity.js#L39

Needs to be changed to

```
options.userId = decodeURIComponent(options.id.substring(Identity.prefixUUID.length));
```

Thanks for bringing this error to our attention.  Will try to get this fixed by end of week.  Ping me if that timeline is a problem.

**Comment by jeffijoe** [comment]
It does not seem to break anything for me so thats fine. 😀

**Comment by jeffijoe** [comment]
It might actually be breaking something after all. Any idea when the fix will be out?

**Comment by michaelkantor** [comment]
Its unfortunately been grouped with a change whose side effects I'm not 100% confident in yet.  I may try and separate them so I can ship this sooner.  But it will be a few days at least.  Or you can try out Branch `3.3.3` if you need.

**Comment by michaelkantor** [comment]
Fix deployed with WebSDK 3.3.3

**Comment by jeffijoe** [comment]
Thank you!
